### PR TITLE
Docs: Documented delayed allocation settings

### PR DESCRIPTION
**Pull request by clintongormley**
Relates to: #11712

### CODE REVIEW

**Comment by clintongormley** [comment]
@kimchy please review

File: docs/reference/index-modules/allocation/delayed.asciidoc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+the master reacts by:
+
+* Promoting a replica shard to primary to replace any primaries that were on the node.
+* Allocating replica shards to replace the missing replicas (assuming there are enough nodes).
+* Rebalancing shards evenly across the remaining nodes.
+
+These actions are intended to protect the cluster against data loss by
+ensuring that every shard is fully replicated as soon as possible.
+
+This ``shard-shuffle'' can, however, put a lot of extra load on the cluster
```
Comment (kimchy) [inline docs/reference/index-modules/allocation/delayed.asciidoc]: I would explain that we do throttling, something like "Even though throttling is done on a node level for concurrent recoveries, as well as across the cluster, this "shard shuffle" can cause extra load on ...."

File: docs/reference/index-modules/allocation/delayed.asciidoc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+------------------------------
+// AUTOSENSE
+
+With delayed allocation enabled, the above scenario changes to look like this:
+
+* Node 5 loses network connectivity.
+* The master promotes a replica shard to primary for each primary that was on Node 5.
+* The master logs a message that allocation of unassigned shards has been delayed, and for how long.
+* The cluster remains yellow because there are unassigned replica shards.
+* Node 5 returns after a few minutes, before the `timeout` expires.
```
Comment (kimchy) [inline docs/reference/index-modules/allocation/delayed.asciidoc]: cluster health also return the number of delayed unassigned shards, maybe mention it?

Comment (clintongormley) [inline docs/reference/index-modules/allocation/delayed.asciidoc]: Mentioned below, under monitoring

**Comment by clintongormley** [comment]
Updated to address comments. Merging to master

**Comment by clintongormley** [comment]
Backported to 1.x in https://github.com/elastic/elasticsearch/commit/985f8ea7a9bdf8b39c47a81759041df593d30fc3

**Pull request merged**